### ISSUE TITLE
ObjC @try @catch cause internal compiler error

### ISSUE THREAD

**Issue by mlfarrell** [Microsoft/WinObjC#160]
Code that craps out:

```
  @try
  {
    [target mirrorSketch];
  }
  @catch(NSException *e)
  {
#ifdef DEBUG
    NSLog(@"Exception: %@", e);
#endif
    NSString *message = NSLocalizedString(@"The generated mesh will be too complex to be efficiently edited by this software.", nil);
    NSString *title = NSLocalizedString(@"Mesh too complex", nil);
    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:title message:message 
                                                   delegate:nil cancelButtonTitle:NSLocalizedString(@"Ok", nil) otherButtonTitles:nil];
    [target restartSketch];
    [alert show];
  }
  @finally
  {  
    [[EAGLView sharedGLView] drawView:nil];  
  }
```

**Comment by PooyaZv** [comment]
Can you please try with ARC turned off? (Project > Properties > Clang > General > Enable Objective-C ARC = No)

Also, please see https://github.com/Microsoft/WinObjC/issues/354

**Comment by DHowett-MSFT** [comment]
Fixed with #2869
